web/server/api/endpoints/files.py

```
@@ -162,18 +187,16 @@ def walk_path(
     )
 
 
-def _get_file_with_content(
-    path: Path,
-    settings: Settings,
-) -> models.File:
+def _get_file_with_content(file_path: Path, path: str) -> models.File:
     """Get a file, including its contents."""
-    file_path = settings.project_path / path
-
-    with open(file_path) as f:
-        content = f.read()
+    try:
+        content = str(file_path.read_text())
+    except Exception:
+        print(f"Unable to read file: {file_path}")
+        content = None
```

Shouldn't we let the error bubble up?